packages/blueprints-integration/src/rundown.ts

```diff
@@ -501,8 +501,13 @@ export interface IBlueprintPieceInstance<TMetadata = unknown> {
 
 	/** If this piece has been created play-time using an AdLibPiece, this should be set to it's source piece */
 	adLibSourceId?: string
-	/** If this piece has been insterted during run of rundown (such as adLibs), then this is set to the timestamp it was inserted */
-	dynamicallyInserted?: Time
+	/** If this piece has been insterted during run of rundown (such as adLibs), then this is set and includes the timestamp when it was inserted */
+	dynamicallyInserted?: {
```

Comment:
`dynamicallyInserted` sounds like it's a boolean. I know you didn't create it but maybe this would be more precise naming?:
```
dynamicallyInsertion: {
  insertedAt: Time,
  isInsertedIntoNextPart: boolean
}
```

Reply:
`dynamicallyInsertion` (or rather `dynamicInsertion`) does not sound right either. Because this is an optional property, it's presence means that the piece has been dynamically inserted. In most cases it might be all that matters So 
`if (pieceInstance.dynamicallyInserted)` reads very naturally, while `pieceInstance.dynamicallyInserted.intoNextPart` and `pieceInstance.dynamicallyInserted.time` also kind of sound like English and wothout repetitions

Reply:
I agree `if(pieceInstance.dynamicallyInserted)` does read very naturally. So naturally that you would expect it to be a boolean, but it's not and hence misleading.
If the argument for it to be named like a boolean is because it is optional, then maybe don't make it optional? 
If you do:
```
dynamicInsertion: {
  insertedAt?: Time,
  isInsertedIntoNextPart: boolean
}
```
Then you could write: `if(!!peiceInstance.dynamicInsertion.insertedAt)` which also reads quite well and it doesn't hint it to be a type that it is not. (But we are getting close to that Law of Demeter would like to have a word with us)

But it's just a suggestion :)